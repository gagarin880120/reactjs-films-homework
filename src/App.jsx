import { hot } from 'react-hot-loader/root';
import React from 'react';
import MovieDetailsPage from './pages/MovieDetailsPage/MovieDetailsPage';

function App() {
  return (
    <>
      <MovieDetailsPage />
    </>
  );
}

export default hot(App);
