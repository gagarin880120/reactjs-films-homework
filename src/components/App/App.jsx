import { hot } from 'react-hot-loader/root';
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import {
  Switch,
  Route,
  Redirect,
  useLocation,
} from 'react-router-dom';
import MovieDetailsPageContainer from '../../pages/MovieDetailsPage';
import SearchResultsPageContainer from '../../pages/SearchResultsPage';
import styles from './App.module.scss';
import Spinner from '../Spinner/Spinner';

function App({
  onAppLoad, genres, onReload, currentAPIRequest,
}) {
  const location = useLocation();
  useEffect(() => {
    if (!genres) {
      onAppLoad();
    }
    if (location.pathname.length > 1) {
      onReload(location.pathname);
    }
  }, [location]);

  return (
    genres
      ? (
        <div className={styles.wrapper}>
          <Switch>
            <Route exact path="/">
              <Redirect to={`/${currentAPIRequest}`} />
            </Route>
            <Route path={`/${currentAPIRequest}`} exact component={SearchResultsPageContainer} />
            <Route path="/movie/:id" exact component={MovieDetailsPageContainer} />
          </Switch>
        </div>
      ) : (
        <div className={styles.loading}>
          <Spinner />
        </div>
      )
  );
}

export default hot(App);

App.propTypes = {
  onAppLoad: PropTypes.func,
  genres: PropTypes.instanceOf(Array),
  currentAPIRequest: PropTypes.string,
  onReload: PropTypes.func,
};

App.defaultProps = {
  onAppLoad: null,
  genres: [],
  currentAPIRequest: '',
  onReload: null,
};
