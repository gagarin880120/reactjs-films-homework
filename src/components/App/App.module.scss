.wrapper {
  font-family: Arial, Helvetica, sans-serif;
  color: #fff;
}

.loading {
  margin-top: 20%;
}
