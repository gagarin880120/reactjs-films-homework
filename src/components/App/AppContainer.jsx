import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getGenres, getMovies } from '../../redux/actions';
import { genresSelector } from '../../redux/selectors';
import getURL from '../../utils/utils';
import App from './App';
import Spinner from '../Spinner/Spinner';
import styles from './App.module.scss';

export function AppContainer({ onAppLoad, genres }) {
  useEffect(() => {
    if (!genres) {
      onAppLoad();
    }
  }, []);

  return (
    genres
      ? <App /> : (
        <div className={styles.loading}>
          <Spinner />
        </div>
      )
  );
}

export const mapStateToProps = (state) => ({
  genres: genresSelector(state),
});

export const mapDispatchToProps = (dispatch) => ({
  onAppLoad() {
    dispatch(getGenres());
    dispatch(getMovies(getURL('trending'), 1));
  },
});

AppContainer.propTypes = {
  onAppLoad: PropTypes.func,
  genres: PropTypes.instanceOf(Array),
};

AppContainer.defaultProps = {
  onAppLoad: null,
  genres: [],
};

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
