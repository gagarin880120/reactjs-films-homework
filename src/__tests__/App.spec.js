import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import App from '../App';

test('App component should render without crashing', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
