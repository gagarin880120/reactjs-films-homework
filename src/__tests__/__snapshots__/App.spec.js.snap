// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without crashing 1`] = `
<React.Fragment>
  <MovieDetailsPage />
</React.Fragment>
`;
